Document a training loop for streaming dataset (#3370)
huggingface/datasets: docs/source/stream.rst

   >>> from datasets import load_dataset
   >>> dataset = load_dataset('m4', 'en', streaming=True, split='train')
   >>> dataset = dataset.remove_columns('timestamp')

``Map``
^^^^^^^

> As for regular map-style :class:`datasets.Dataset`, 🤗 Datasets features :func:`datasets.IterableDataset.map` to process your dataset.

Reviewer: ```suggestion
Similar to the :func:`datasets.Dataset.map` function for a regular :class:`datasets.Dataset`, 🤗  Datasets features :func:`datasets.IterableDataset.map` for processing :class:`datasets.IterableDataset`s.
```

> Unlike a map-style dataset though, the processing is applied on-the-fly when the examples are streamed.

Reviewer: ```suggestion
:func:`datasets.IterableDataset.map` applies processing on-the-fly when examples are streamed.
```

It allows you to apply a processing function to each example in a dataset, independently or in batches. This function can even create new rows and columns.

> In the following example, you will apply tokenization to the dataset. The function needs to accept and output a ``dict``:

Reviewer: ```suggestion
The following example demonstrates how to tokenize a :class:`datasets.IterableDataset`. The function needs to accept and output a ``dict``:
```

.. code-block::

   >>> from datasets import load_dataset
   >>> from transformers import AutoTokenizer
   >>> dataset = load_dataset("mc4", "en", streaming=True, split="train")
   >>> tokenizer = AutoTokenizer.from_pretrained('distilbert-base-uncased')
   >>> def encode(examples):
   ...     return tokenizer(examples['text'], truncation=True, padding='max_length')
   >>> dataset = dataset.map(encode, batched=True)
   >>> next(iter(dataset))
   {'input_ids': 101, 8466, 1018, 1010, 4029, 2475, 2062, 18558, 3100, 2061, ...,1106, 3739, 102],
   'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ..., 1, 1]}

> In a training loop

Reviewer: ```suggestion
Stream in a training loop
```

^^^^^^^^^^^^^^^^^^

> First let's shuffle the dataset:

Reviewer: ```suggestion
:class:`datasets.IterableDataset`s can be integrated into a training loop. First, shuffle the dataset:
```
